main.go

```diff
@@ -37,14 +41,31 @@ func main() {
 		return
 	}
 
-	if len(os.Args) == 2 {
-		ioffice.CreateReservation(me, roomID, dateparse.MustParse(os.Args[1]))
-	}
+	if len(os.Args) < 2 {
+		ioffice.ListReservations()
+	} else {
 
-	if len(os.Args) == 3 {
-		room := ioffice.GetRoom(os.Args[2])
-		ioffice.CreateReservation(me, room.ID, dateparse.MustParse(os.Args[1]))
+		switch os.Args[1] {
+		case "list":
+			ioffice.ListReservations()
+		case "create":
+			if len(os.Args) == 2 {
+				usage()
+			}
+			if len(os.Args) == 3 {
+				ioffice.CreateReservation(me, roomID, dateparse.MustParse(os.Args[2]))
+			}
+			if len(os.Args) == 4 {
+				room := ioffice.GetRoom(os.Args[2])
+				ioffice.CreateReservation(me, room.ID, dateparse.MustParse(os.Args[2]))
+			}
+		case "checkin":
+			reservationID := os.Args[2]
+			ioffice.CheckIn(reservationID)
+		case "cancel":
+			reservationID := os.Args[2]
+			ioffice.CancelReservation(reservationID)
+			usage()
```

Comment:
are you missing a `default:` before this `usage()`? I presume the goal is to respond with usage if an unknown command is given

Reply:
Yeah spotted that, thank you :)